api/queries_repo.go

```diff
@@ -524,6 +525,35 @@ func ForkRepo(client *Client, repo ghrepo.Interface, org string) (*Repository, e
 	}, nil
 }
 
+func LastCommit(client *Client, repo ghrepo.Interface) (*git.Commit, error) {
+	var responseData struct {
+		Repository struct {
+			DefaultBranchRef struct {
+				Target struct {
+					Commit struct {
+						History struct {
+							Edges []struct {
+								Node struct {
+									Sha   string `graphql:"oid"`
+									Title string `graphql:"messageHeadline"`
+								}
+							}
+						} `graphql:"history(first:1)"`
+					} `graphql:"... on Commit"`
+				}
+			}
+		} `graphql:"repository(owner: $owner, name: $repo)"`
+	}
+	variables := map[string]interface{}{
+		"owner": githubv4.String(repo.RepoOwner()), "repo": githubv4.String(repo.RepoName()),
+	}
+	gql := graphQLClient(client.http, repo.RepoHost())
+	if err := gql.QueryNamed(context.Background(), "LastCommit", &responseData, variables); err != nil {
+		return nil, err
+	}
+	return (*git.Commit)(&responseData.Repository.DefaultBranchRef.Target.Commit.History.Edges[0].Node), nil
```

Comment:
It feels super weird to cast at unrelated struct into a `git.Commit` object in Go. I would rather copy the fields manually:
```go
return &git.Commit{
  Sha: ...,
  Title: ...
}, ...
```